opm/simulators/utils/PartiallySupportedFlowKeywords.hpp

```diff
@@ -0,0 +1,104 @@
+/*
+  Copyright 2021 Equinor.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef OPM_PARTIALLYSUPPORTEDFLOWKEYWORDS_HEADER_INCLUDED
+#define OPM_PARTIALLYSUPPORTEDFLOWKEYWORDS_HEADER_INCLUDED
+
+
+#include <map>
+#include <opm/parser/eclipse/Parser/ErrorGuard.hpp>
+#include <opm/parser/eclipse/Parser/ParseContext.hpp>
+#include <opm/simulators/flow/KeywordValidation.hpp>
+
+/*
+    Here keywords are defined that are supported by flow, but have options that
+    are only partially supported.
+
+    The keywords are specified in a mapping with the keyword names as keys, and
+    values that describe the set of supported options. These are described by a
+    mapping from the option name to a struct of properties, defined in KeywordValidation.hpp.
+
+    This struct has the following fields:
+
+    critical (bool) : if this is a critical error.
+    permitted_values (vector of strings) : the list of values that is allowed.
+    message (optional string): an optional message to add to the error reported by flow.
+
+    Below is the set of partiall supported keywords, currently used by flow.
+*/
+
+namespace Opm::FlowKeywordValidation
+{
+
+
+const Opm::KeywordValidation::PartiallySupportedKeywords<std::string> partially_supported_kewyords_strings = {
+    {
+        "COMPORD",
+        {
+            {
+                "ORDER_TYPE",
```

Comment:
Rather than item name, it is better to use item number here. That way we will also avoid having to rewrite the item names in the json schemes.

Reply:
I guess this code is interesting for three different groups of people:

1. Pure developers who implement and maintain the solution.
2. Intermediate developers/users who might maintain the list of supported/unsupported.
3. End users.

At least for categories 1 & 2 I think string identifiers is better than integers, but we can of course present the user with an integer number even if we use strings as internal ID.

Reply:
I believe it is group 2 we want to cater to. And currently they can all check the manual for item numbers, or just look at the data deck at hand. String identifiers is another beast, especially since they are different in the Json schemes and the manual.